src/Transport/Config/ExtensionPoints/AzureServiceBusNamespacePartitioningSettings.cs

```diff
@@ -29,7 +29,7 @@ public AzureServiceBusNamespacePartitioningSettings(SettingsHolder settings)
             return this;
         }
 
-        public AzureServiceBusNamespacePartitioningSettings AddNamespace(string name, string connectionString)
+        public AzureServiceBusNamespacePartitioningSettings AddNamespace(string name, string connectionString, NamespacePurpose purpose = NamespacePurpose.Partitioning)
```

Comment:
The config API is a bit weird here, IMO: we are into "partitioning" config section but, as a user, I can register a `namepsace` for `DestinationOnly` purpose.
On the other side, to register a namespace for `DestinationOnly` purpose, I have to go through "partitioning" configuration to find the API